crypto/threads_pthread.c

```diff
@@ -162,6 +166,12 @@ int CRYPTO_atomic_add(int *val, int amount, int *ret, CRYPTO_RWLOCK *lock)
         *ret = __atomic_add_fetch(val, amount, __ATOMIC_ACQ_REL);
         return 1;
     }
+# elif defined(__sun) && (defined(__SunOS_5_10) || defined(__SunOS_5_11))
+    /* This will work for all future Solaris versions. */
+    if (ret != NULL) {
```

Comment:
Not sure this is strictly necessary since we don't do this test anywhere else in this function.

Reply:
The `if` was added to preserve the original behaviour (GCC/#else) and avoid "unreachable code" warning from the compiler.